Assets/MRTK/Providers/WindowsMixedReality/XR2018/WindowsMixedRealityDeviceManager.cs

```diff
@@ -671,7 +717,14 @@ private BaseWindowsMixedRealitySource GetOrAddController(InteractionSource inter
             }
 
             string nameModifier = controllingHand == Handedness.None ? interactionSource.kind.ToString() : controllingHand.ToString();
+
+#if HP_CONTROLLER_ENABLED
+            bool isHPController = !interactionSource.supportsTouchpad;
+            string inputSourceName = isHPController ? $"HP Motion Controller {nameModifier}" : $"Mixed Reality Controller {nameModifier}";
+            var inputSource = Service?.RequestNewGenericInputSource(inputSourceName, pointers, inputSourceType);
+#else
             var inputSource = Service?.RequestNewGenericInputSource($"Mixed Reality Controller {nameModifier}", pointers, inputSourceType);
+#endif
```

Comment:
It don't feel like there's any dependency on `HP_CONTROLLER_ENABLED` to set the name of the input source or set the bool. Can this just be the "normal" path (basically remove the `#if` and the `#else` case)?

Reply:
We can do that, the guard was here mostly because the pipeline doesn't correctly report the touchpad status at the moment, so all WMR controllers will end up being HP controllers was the main issue.

Reply:
This API should report touchpad properly; I thought it was just XR SDK that had the issue